util/repos/repos.go

```diff
@@ -0,0 +1,125 @@
+package repos
+
+import (
+	"io/ioutil"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// Client is a generic git client interface
+type Client interface {
+	// Return the directory of the checkout code. Aka "working tree".
```

Comment:
If we decided to introduce set of separate APIs then first class helm support we would have two separate code paths where one needs only git repos and another needs only helm repos. So it almost defeats benefit from config refactoring. I realize this is pretty big part of this PR but it would be good to avoid unnecessary config migration.